### ISSUE TITLE
Maps UI accessibility improvements

### ISSUE THREAD

**Issue by Malvoz** [duckduckgo/community-platform#1531]
[DuckDuckGo maps](https://help.duckduckgo.com/duckduckgo-help-pages/features/maps/) has a few accessibility issues. Additionally it inherits [accessibility issues in MapKit](https://github.com/Malvoz/web-maps-wcag-evaluation/blob/master/README.md) (which I have reported in the [Apple developer forum](https://developer.apple.com/forums/thread/674971)).

I believe the following issues are specific to DuckDuckGo maps (not inherited by the MapKit JS framework):

<details><summary>Back button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109389667-9bc75600-790d-11eb-9f11-ddfdea76cae4.png"><br><br>

- Is missing `role="button"` (preferably it should be `<button>`)
- Is missing an accessible name e.g. `aria-label="Back"`

<hr>

</details>

<details><summary>Clear search button</summary><br>
<img width="450"  alt  src="https://user-images.githubusercontent.com/26493779/109389020-cebc1a80-790a-11eb-8913-d0ac0264d5f5.png"><br><br>

- The accessible name is "X". This should be descriptive, e.g. `aria-label="Clear search"`
- Although fousable, it is not _reachable_ using a keyboard

<hr>

</details>

<details><summary>Search button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109389027-d2e83800-790a-11eb-9f75-c7190c4d4fb1.png"><br><br>

- The accessible name is "S". This should be descriptive, e.g. `aria-label="Search"`

<hr>

</details>

<details><summary>Close sidebar button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109389049-f1e6ca00-790a-11eb-8723-9b0e52ef0c42.png"><br><br>

- The accessible name is "<". This should be descriptive, e.g. `aria-label="(Close|Hide) details"`

<hr>

</details>

<details><summary>Open sidebar button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109389057-f6ab7e00-790a-11eb-9b71-596b70758758.png"><br><br>

- The accessible name is ">". This should be descriptive, e.g. `aria-label="(Open|Show) details"`

<hr>

</details>

<details><summary>Locate Me button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109389035-ded3fa00-790a-11eb-8b43-a13bed12fb7e.png"><br><br>

- Is not keyboard focusable
- Is missing semantics. It should be `<button>` (or `<a (tabindex="0"|href="#") role="button">`)
- Is missing an accessible name. This should be descriptive, e.g. `aria-label="Locate me"`

<hr>

</details>

<details><summary>Close Locate Me button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109389066-ff9c4f80-790a-11eb-9c60-1e70a39a38ee.png"><br><br>

- Is not keyboard focusable
- Has no semantics. It should be `<button>` (or `<a (tabindex="0"|href="#") role="button">`)
- Is missing an accessible name. This should be descriptive, e.g. `aria-label="Close"`

<hr>

</details>

<details><summary>Activate Locate Me button</summary><br>
<img width="450"  alt src="https://user-images.githubusercontent.com/26493779/109390899-93264e00-7914-11eb-848c-dbed94470ade.png"><br><br>

- Is not keyboard focusable
- It should be `<button>` (or `<a (tabindex="0"|href="#") role="button">`)

(these things apply to the "button" next to it aswell)

<hr>

</details>

I'm just scratching the surface though, other things to consider:
- Focus delegation (For example, you'd probably want to send focus to the "dialog" after clicking the Locate Me button. As well as send focus back to the Locate Me button once the dialog has been closed. Per the <span title="Web Accessibility Initiative - Accessible Rich Internet Applications">WAI-ARIA</span> Authoring Practices recommended [Keyboard Interactions](https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-3) for [buttons](https://www.w3.org/TR/wai-aria-practices/#button))
- CSS outlines. It seems outlines are disabled, consider enabling outlines at least for keyboard users (see [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible)).
